tests/controls/textctrltest.cpp

```diff
@@ -260,51 +276,109 @@ void TextCtrlTestCase::ReadOnly()
 void TextCtrlTestCase::MaxLength()
 {
 #if wxUSE_UIACTIONSIMULATOR
+    EventCounter maxlen(m_text, wxEVT_TEXT_MAXLEN);
+
+    m_text->SetFocus();
+    wxYield();
+
+    wxUIActionSimulator sim;
+
+    if ( ms_style == wxTE_MULTILINE )
+    {
+#if defined(__WXMSW__) || defined(__WXQT__)
+        m_text->SetMaxLength(250);
+
+        const wxString linePattern = MakeLinePattern();
+
+        m_text->AppendText(wxString::Format("%s\n", linePattern));
+        m_text->SelectAll();
+        m_text->Copy();
+
+        m_text->SetInsertionPointEnd();
+
+        sim.Char('v', wxMOD_CONTROL); // Paste copied line.
+        wxYield();
+
+        CPPUNIT_ASSERT_EQUAL(0, maxlen.GetCount());
+
+        sim.Char('v', wxMOD_CONTROL); // Paste copied line (2nd time).
+        wxYield();
+
+        CPPUNIT_ASSERT_EQUAL(1, maxlen.GetCount()); // Maximum length reached.
+        maxlen.Clear();
+
+        sim.Text("7"); // Should be rejected.
+        wxYield();
+
+        CPPUNIT_ASSERT_EQUAL(1, maxlen.GetCount());
+        maxlen.Clear();
+
+        // Depending on the underlying system, the new line (NL) could be
+        // LF, CR or CRLF, and as a consequence the length of the 1st and
+        // 2nd lines should be 100 + sizeof(NL) each, and the length of the
+        // last line should be 50 - 2*NL.
```

Comment:
@AliKet ,
Did you check the MSW behavior for the new line character?

Reply:
The comments above describes what the control does internally, i.e. new line characters are also considered and counted by the control to decide whether the maximum length is reached or not. For example, if you have a control with MAXLEN set to 500 with 5 lines of text, and each line contains 100 chars, the last line will always have 100 - 4* sizeof(NL).
- If NL is just a LF or CR the result is 96 chars.
- If NL is CRLF the result is 92 chars.